### PR TITLE
fix: broken devtools with dynamic path

### LINKED ISSUE #176: NextJS dynamic urls in root breaking transaction button
### Describe the bug

Will copy paste alot of stuff from the wevm discord thread https://discord.com/channels/1156791276818157609/1220449473575260232:

What i'm trying to create is a dynamic frame where I can pull in a variable from the URL and change the frame as I see fit. e.g. localhost:3000/api/variable

it loads the metadata and dev environment  e.g.  localhost:3000/api/variable/dev 

but when i try and send a transaction to a contract i get a error code 400 (invalid url). the transaction works fine if i have the root url as '/' rather than '/:variable', so the issue isn't the transaction frame.

Here is a Warpcast dev tool link that showcases the issue in action:

https://warpcast.com/~/developers/frames?url=https%3A%2F%2Fminimal-reproduction-frog.vercel.app%2Fapi%2F5

@ dalechyn in the wevm discord suggested it might be hono routehandling/priority, so I swapped out the root frame and the transaction frame, but that didn't seem to fix the issue.
https://github.com/Seranged/minimal-reproduction-frog/blob/main/app/api/%5B%5B...routes%5D%5D/route.tsx
image

Thanks for your guys' help [🫡](https://emojipedia.org/saluting-face)

### Link to Minimal Reproducible Example

https://github.com/Seranged/minimal-reproduction-frog

### Steps To Reproduce

Create a nextjs application that has a dynamic url in the root frame e.g.

 `app.frame('/:dynamicUrl', (c) => {`

Create a transaction frame 

`app.transaction('/transaction', async (c) => {`

Try perform the transaction on a deployed build

### Frog Version

0.7.4

### TypeScript Version

^5

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/frog/issues) for the bug I encountered.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by ggomaeng**
Added new catch all route in `frog-base.tsx` so it's the first route prioritized to be used for devtools when `this_dev` is set. If `this_dev` is not set, it just relays the request.

Since devtools is registered at the end of all the route definition to have information about existing roots, having any dynamic path before devtools will have higher priority over the devtools path & devtools will become obsolete.

This fixes #176 - this problem exists not only in Next.js, but also in other environments as well.

Tested on my end and everything seems to work! including transactions :+1:

### Minimal reproduction repo
https://github.com/ggomaeng/frog-dynamic-route-minimal-reproduction

1. clone the repo & install dependencies
2. npm run dev
3. go to `http://localhost:3000/api/dev`
4. devtools doesn't show. just white blank screen

### To test this PR
1. clone the above repo
2. build this PR and link frog in the repo
3. go to `http://localhost:3000/api/dev`
4. devtools work as expected

---

### Steps To Reproduce

1. Create a Next.js frog application that has a dynamic url in the root frame e.g.
```ts
// app/api/[[..routes]]/route.tsx
/** @jsxImportSource frog/jsx */

import { Button, Frog, TextInput } from "frog";
import { devtools } from "frog/dev";
import { handle } from "frog/next";
import { serveStatic } from "frog/serve-static";

const app = new Frog({
  basePath: "/api",
});

app.frame("/", (c) => {
  const { buttonValue, inputText, status } = c;
  const fruit = inputText || buttonValue;
  return c.res({
    image: (
      <div
        style={{
          alignItems: "center",
          background:
            status === "response"
              ? "linear-gradient(to right, #432889, #17101F)"
              : "black",
          backgroundSize: "100% 100%",
          display: "flex",
          flexDirection: "column",
          flexWrap: "nowrap",
          height: "100%",
          justifyContent: "center",
          textAlign: "center",
          width: "100%",
        }}
      >
        <div
          style={{
            color: "white",
            fontSize: 60,
            fontStyle: "normal",
            letterSpacing: "-0.025em",
            lineHeight: 1.4,
            marginTop: 30,
            padding: "0 120px",
            whiteSpace: "pre-wrap",
          }}
        >
          {status === "response"
            ? `Nice choice.${fruit ? ` ${fruit.toUpperCase()}!!` : ""}`
            : "Welcome!"}
        </div>
      </div>
    ),
    intents: [
      <TextInput placeholder="Enter custom fruit..." />,
      <Button value="apples">Apples</Button>,
      <Button value="oranges">Oranges</Button>,
      <Button value="bananas">Bananas</Button>,
      <Button.Transaction target="/transaction">hello</Button.Transaction>,
      status === "response" && <Button.Reset>Reset</Button.Reset>,
    ],
  });
});

app.frame("/:dynamicUrl", (c) => {
  const dynamicUrl = c.req.param("dynamicUrl");
  return c.res({
    image: (
      <div
        style={{
          height: "100vh",
          width: "100vw",
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          backgroundColor: "#000000",
        }}
      >
        <div
          style={{
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
            color: "white",
            fontSize: 60,
          }}
        >
          Test Minimal Dynamic Url ID: {dynamicUrl}
        </div>
      </div>
    ),
    intents: [
      <Button>Hello World!</Button>,
    ],
  });
});

devtools(app, { serveStatic });
export const GET = handle(app);
export const POST = handle(app);
```

### Frog Version

0.7.13

---

cc @Seranged, @christopherwxyz, @chuckstock.

### CODE REVIEW

**Comment by ggomaeng** [comment]
Actually it looks like more testing has to be done - it seems like this breaks a lot of functionalities.

**Comment by ggomaeng** [comment]
Actually it looks like more testing has to be done - it seems like this breaks a lot of functionalities.

**Comment by ggomaeng** [comment]
~~I am building frog locally to test using pnpm build & link, but transaction seems to not work with this PR. I forked the main branch. Is my code breaking the transaction functionality, or did I set up local dev environment wrong? Any pointers would be appreciated if something looks wrong!~~

Update: Transaction seems to work after clean fork of the frog repo and building on top of it.

**Comment by ggomaeng** [comment]
Edit: resolved wrong route priority in my minimal reproduction code :') 

~~Seems like inputText & buttonValues are working but GET /image paths are not loading too. I swear the images were working before this PR 🤔~~

image

~~Seems like the content type returns text/html; charset=UTF-8~~

**Comment by ggomaeng** [comment]
@tmm do you think I am on the right track with this solution?

**Comment by ggomaeng** [comment]
Realized my example demo had wrong route priority so the images weren't rendering 😅

image

**Comment by ggomaeng** [comment]
Minimal repo used for testing: https://github.com/ggomaeng/frog-dynamic-route-minimal-reproduction

### To test this PR
1. clone the above repo
2. build this PR and link frog in the repo
3. go to `http://localhost:3000/api/dev`

**Review by dalechyn — APPROVED** [review]
tested it and confirm that the patch solves the issue.

File: src/frog-base.tsx
Comment (tmm) [inline src/frog-base.tsx:1]: Would it make more sense to add this to `devtools.tsx`?

File: .changeset/silver-plums-repeat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"frog": patch
+---
+
+Fixed the `this._dev` route to be prioritized over dynamic path (`/:dynamic-path`) if it exists
```
Comment (tmm) [inline .changeset/silver-plums-repeat.md]:
```suggestion
Prioritized devtools route over dynamic path if it exists.
```

File: src/frog-base.tsx
Comment (ggomaeng) [reply src/frog-base.tsx:1]: That was my initial approach - trying not to touch `frog-base.tsx` at all and trying tweaking the `devtools.tsx`.

However I could not figure out a solution for overriding the existing hono route priority without making things more complicated.

**Pull request merged**